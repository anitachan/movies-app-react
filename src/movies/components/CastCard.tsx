import { Box, Button, Card, CardMedia, Typography } from '@mui/material';
import { Cast } from '../models/cast.model';
import noUser from '../../assets/no-user.png';
import { useNavigate } from 'react-router-dom';

interface Props {
  actor: Cast;
}
export const CastCard = ({ actor }: Props) => {
  const url = process.env.REACT_APP_IMAGES_URL;
  const profile_path = actor.profile_path ? `${url}${actor.profile_path}` : noUser;
  const navigate = useNavigate();

  const handleClick = () => {
    navigate(`/actor/${actor.id!}`);
  };

  return (
    <Button onClick={handleClick}>
      <Box sx={{ maxWidth: 250, textAlign: 'center' }}>
        <Card sx={{ maxWidth: 250 }}>
          <CardMedia
            component="img"
            src={profile_path}
            alt={actor.name}
            sx={{
              width: '100%',
              height: 350,
              objectFit: 'cover',
              borderRadius: 2,
            }}
          />
        </Card>
        <Typography variant="body2" fontWeight="bold" noWrap>
          {actor.name}
        </Typography>
        <Typography variant="caption" color="text.secondary" noWrap>
          {actor.character}
        </Typography>
      </Box>
    </Button>
  );
};
