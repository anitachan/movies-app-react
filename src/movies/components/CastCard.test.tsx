import { fireEvent, render, screen } from '@testing-library/react';
import { CastCard } from './CastCard';
import { fixtureOf } from 'ts-mock-autofixture-kit';
import { Cast } from '../models/cast.model';
import noUser from '../../assets/no-user.png';

const mockUseNavigate = jest.fn();
jest.mock(
  'react-router-dom',
  () => ({
    useNavigate: () => mockUseNavigate,
  }),
  { virtual: true },
);

describe('CastCard', () => {
  const mockActor = fixtureOf<Cast>({ seed: 42 }).create();

  test('CreditsCard matches snapshot', () => {
    const { container } = render(<CastCard actor={mockActor} />);

    expect(container).toMatchSnapshot();
  });

  test('should redirect when movie is clicked', () => {
    render(<CastCard actor={mockActor} />);
    const button = screen.getByRole('button');
    fireEvent.click(button);

    expect(mockUseNavigate).toHaveBeenCalledWith(`/actor/${mockActor.id!}`);
  });

  test('should set noUser when poster is empty', () => {
    const mockActorPoster = fixtureOf<Cast>()
      .with('profile_path', () => null)
      .with('name', () => 'No Poster Actor')
      .create();

    render(<CastCard actor={mockActorPoster} />);
    const image = screen.getByAltText('No Poster Actor') as HTMLImageElement;

    expect(image).toBeInTheDocument();
    expect(image.src).toContain(noUser);
  });
});
