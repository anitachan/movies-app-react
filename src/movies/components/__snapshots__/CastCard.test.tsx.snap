// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CastCard CreditsCard matches snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiBox-root css-13c7vug"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1k4oiox-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <img
          alt="3CpplugYH0Te"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-11jcf76-MuiCardMedia-root"
          src="https://image.tmdb.org/t/p/originalaGTqxm8FGpgY"
        />
      </div>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiTypography-noWrap css-1ydkb93-MuiTypography-root"
      >
        3CpplugYH0Te
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-noWrap css-fgvunj-MuiTypography-root"
      >
        Hi1CDBoab94W
      </span>
    </div>
  </button>
</div>
`;
